PR#7146, manual: use the in-tree compiler and interpreter for building the manual

Octachron opened the pull request:
The aim of this PR is to make the manual build process uses the in tree compiler and interpreter. 

Currently, the manual uses the system compiler for building its tools (in `manual/tools`) and 
the system interpreter for evaluating the code within the `caml_example` and `caml_eval` environment.

Using the system interpreter is fine most of the time, since these environments are used only in the tutorial part of the manual which covers a very stable part of the language. However, there are significant exceptions: newly deprecated functions. Ideally, the manual should never contain deprecation warning in its generated examples; but a mismatch between the manual interpreter and in-tree interpreter can hide such warnings. Ensuring that the manual is always build with the in-tree version of the interpreter would help to detect these warnings earlier.

Moreover, if the `caml_eval` and `caml_example` environment were to be interpreted with the in-tree interpreter, it will be easier to use them in the extension section of the manual. This would alleviate the need to write hand-written latex code mimicing the result of the `caml_example` environment. 

This PR achieves its objective by splitting the `testsuite/makefiles/Makefile.common` makefile in two parts: 
- `Makefile.tools` contains the definitions of the in-tree compiler and interpetrer variables
- `Makefile.common` includes `Makefile.tools` and defines the same targets as previously.

Then, the new makefile `Makefile.tools` is included within the manual tools makefile and the tutorial
makefile to
- compile the manual tools with the in-tree compiler
- eval the caml_example and caml_eval environment with the in-tree interpreter

Octachron committed:
Manual: makefiles use in-tree compiler/interpreter

First, this commit splits testsuite/makefiles/Makefile.common in two:
* Makefile.tools contains the definitions of the in-tree compiler and
interpreter variables
* Makefile.common includes Makefile.tools and defines the same targets
  as previously.

Second, this commit uses the definitions of the in-tree compiler and
interpreter in Makefile.tools to:
* compile the manual tools (inside manual/tools)
* eval the caml_example and caml_eval environment in the tutorial part
  of the manual

damiendoligez commented:
I don't like the idea that the manual depends on a file buried in a subdirectory of `testsuite`. IMO `Makefile.tools` should be lifted to the root directory.

Octachron committed:
Manual: move Makefile.tools to root directory

Since, Makefile.tools is used by both the testsuite and the manual, it
should be moved to the first common ancestor of the two subdirectories:
the root directory.

Note: also remove testsuite/lib/empty and use noinit when invoking the
ocaml interpreter.

Octachron commented:
I agree that the location of `Makefile.tools` should be more symmetric with respect to the testsuite and manual paths. For now, I have moved it to the root directory as suggested.

I have also replaced the use of `testsuite\lib\empty` as a dummy ocamlinit file by the the `-noinit` option. Another possibility would be to also move this empty file to the root directory.
